### PR TITLE
DHFPROD-7838: Panned graph view coordinates are not persisted

### PR DESCRIPTION
**Pull request by AkshayBajajML**
#### Checklist: 
```diff
- Note: do not change the below
```

-  ##### Owner:

- [x] JIRA_ID included in all the commit messages
- [x] PR title is in the format JIRA_ID:Title
- [x] Rebase the branch with upstream
- [x] Squashed all commits into a single commit
- [x] Code passes ESLint tests
- N/A Added Tests
- [x] Run UI tests
  

- ##### Reviewer:

- N/A Reviewed Tests

### CODE REVIEW

File: marklogic-data-hub-central/ui/src/components/modeling/graph-view/graph-vis/graph-vis.tsx [hunk 1]
```diff
@@ -571,25 +571,16 @@ const GraphVis: React.FC<Props> = (props) => {
           setCoords(newCoords);
           props.saveEntityCoords(nodes[0], positions.x, positions.y);
         }
+      } else {
+        let ids = Object.keys(coords);
+        // Take current positions and update with drag deltas
+        ids.forEach(id => {
+          let newX = coords[id].graphX + event.event.deltaX;
+          let newY = coords[id].graphY + event.event.deltaY;
+          props.saveEntityCoords(id, newX, newY); // Save to db
```
Comment (wooldridge) [inline marklogic-data-hub-central/ui/src/components/modeling/graph-view/graph-vis/graph-vis.tsx:580]: @AkshayBajajML, right now we're doing an endpoint call for every node that is panned. @xnikhil08 added the ability to update the coordinates of all the entities with a single endpoint call, see here: https://github.com/marklogic/marklogic-data-hub/pull/6105/files#diff-b974bd3271e6146b0ccee23fc50c94a1586ac7f747951f06f25d8462e1c53a4dR129

Can we do the same when reacting to a pan? Build and updatedEntityDefinitions array and then make a updateSavedEntity() call?

**Review by wooldridge — CHANGES_REQUESTED** [review]

File: marklogic-data-hub-central/ui/src/components/modeling/graph-view/graph-vis/graph-vis.tsx (hunk 1 above)
Comment (AkshayBajajML) [reply marklogic-data-hub-central/ui/src/components/modeling/graph-view/graph-vis/graph-vis.tsx]: Thanks! for review and comment.
Yeah that is a good idea. I have implemented the mentioned changes and PR is ready for review again.

**Review by aniamy — DISMISSED** [review]

File: marklogic-data-hub-central/ui/src/components/modeling/graph-view/graph-vis/graph-vis.tsx [hunk 2]
```diff
@@ -571,25 +571,21 @@ const GraphVis: React.FC<Props> = (props) => {
           setCoords(newCoords);
           props.saveEntityCoords(nodes[0], positions.x, positions.y);
         }
+      } else {
+        if (props.entityTypes) {
+          let newCoords = {};
```
Comment (wooldridge) [inline marklogic-data-hub-central/ui/src/components/modeling/graph-view/graph-vis/graph-vis.tsx:576]: @AkshayBajajML, in the pan case we aren't updating the coords state with setCoords, I think we can remove the newCoords init here and also line 579 below?

**Review by wooldridge — CHANGES_REQUESTED** [review]

File: marklogic-data-hub-central/ui/src/components/modeling/graph-view/graph-vis/graph-vis.tsx (hunk 2 above)
Comment (AkshayBajajML) [reply marklogic-data-hub-central/ui/src/components/modeling/graph-view/graph-vis/graph-vis.tsx]: Done

**Review by wooldridge — APPROVED** [review]
Working fine for me, and the code looks good!

**Review by briantang — APPROVED** [review]
working for me as well

File: marklogic-data-hub-central/ui/src/components/modeling/graph-view/graph-vis/graph-vis.tsx [hunk 3]
```diff
@@ -571,25 +571,19 @@ const GraphVis: React.FC<Props> = (props) => {
           setCoords(newCoords);
           props.saveEntityCoords(nodes[0], positions.x, positions.y);
         }
+      } else {
+        if (props.entityTypes) {
+          let updatedEntityDefinitions: any = [];
+          props.entityTypes.forEach(ent => {
+            let entityDefinition: EntityModified = getDefinitionsPayload(ent, {graphX: coords[ent.entityName].graphX + event.event.deltaX, graphY: coords[ent.entityName].graphY + event.event.deltaY});
```
Comment (xnikhil08) [inline marklogic-data-hub-central/ui/src/components/modeling/graph-view/graph-vis/graph-vis.tsx:578]: @AkshayBajajML - Just wondering if **coords** will always get the updated coordinates at new drag location, without using network.getPositions(). Please confirm this.

**Review by xnikhil08 — CHANGES_REQUESTED** [review]

File: marklogic-data-hub-central/ui/src/components/modeling/graph-view/graph-vis/graph-vis.tsx (hunk 3 above)
Comment (wooldridge) [reply marklogic-data-hub-central/ui/src/components/modeling/graph-view/graph-vis/graph-vis.tsx:578]: @xnikhil08 in my investigation, getPositions() did not return updated coordinates when panning (unlike when an individual mode is moved). That's why we have to use the delta values (plus the previous positions as saved in the coords) to get the updated coords to be saved during a pan.

**Review by xnikhil08 — APPROVED** [review]

**Review by wooldridge — APPROVED** [review]
Looks good.

**Pull request merged**